pkg/httputil/logging.go

```diff
@@ -53,13 +54,23 @@ func RequestID(r *http.Request) (*http.Request, string) {
 	return r, reqID
 }
 
+func SourceIP(r *http.Request) string {
+	sourceIP, sourcePort, err := net.SplitHostPort(r.RemoteAddr)
+
+	if err != nil {
+		return "Could not get IP address" //not sure of the error handling here
```

Comment:
Not sure why we need to parse the string here: we either reconstruct soon-ish the same string on success, or return an error during that will be less useful than the original string.

What would we lose if we dropped this function and just used r.RemoteAddr directly?

Reply:
Using just r.RemoteAddr will return the following (tested on localhost in my machine):
```
[::1]:60046
```
Notice the weird brackets? As a solution to that the go team proposes using the net.SplitHostPort function that returns the IP and the Port separately. We can just use the IP that is returned by net.SplitHostPort which in this case would be ``` ::1 ```.  But if you're ok with the brackets thing then we should get rid of the whole SourceIP function and just use r.RemoteAddr.

Reply:
Yikes!  Looks like a classic Go move :-/

I'm fine with brackets.  As yet another alternative, net.GetHostPort seems to return [really neat error messages](https://go.dev/play/p/hqqRs8WlF2h) -- as that example shows, we could use _those_.

```suggestion
		return err.Error()
```